did_doc_builder/src/schema/types/uri.rs

```diff
@@ -0,0 +1,89 @@
+use std::{
+    fmt::{self, Display, Formatter},
+    str::FromStr,
+};
+
+use serde::{Deserialize, Serialize};
+
+use crate::error::DidDocumentBuilderError;
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize, Default)]
+pub struct Uri(uniresid::Uri);
+
+impl Uri {
+    pub fn new(uri: &str) -> Result<Self, DidDocumentBuilderError> {
+        Ok(Self(uniresid::Uri::try_from(uri).map_err(|e| {
+            DidDocumentBuilderError::InvalidInput(format!("Invalid URI: {}", e))
+        })?))
```

Comment:
I think it would be cleaner and more readable to separate the `uniresid::Uri` construction into a variable.